pkg/controllers/user/controllers.go

```diff
@@ -104,10 +107,12 @@ func RegisterFollower(ctx context.Context, cluster *config.UserContext, kubeConf
 }
 
 func RegisterUserOnly(ctx context.Context, cluster *config.UserOnlyContext) error {
-	if err := createUserClusterCRDs(ctx, cluster); err != nil {
+	err := createUserClusterCRDs(ctx, cluster)
+	if err != nil {
 		return err
 	}
 
+	err = settingprovider.Register(ctx, cluster)
```

Comment:
We should return error if err is not nil here.

Reply:
updated.